### PR TITLE
Add `t` method to provide formatted message translation by key

### PR DESCRIPTION
**Pull request by elwayman02**
https://github.com/yahoo/ember-intl/issues/243

### CODE REVIEW

File: addon/services/intl.js [hunk 1]
```diff
@@ -52,6 +52,10 @@ const IntlService = Service.extend(Evented, {
   formatTime: formatterProxy('time'),
   formatDate: formatterProxy('date'),
 
+  t(...args) {
+    return this.formatMessage(this.findTranslationByKey(...args));
```
Comment (jasonmit) [inline addon/services/intl.js]: formatMessage takes 3 arguments: the translation, the value hash, and the format hash

File: addon/services/intl.js [hunk 2]
```diff
@@ -52,6 +52,10 @@ const IntlService = Service.extend(Evented, {
   formatTime: formatterProxy('time'),
   formatDate: formatterProxy('date'),
 
+  t(...args) {
```
Comment (jasonmit) [inline addon/services/intl.js]: Can you cover with some unit tests

Comment (elwayman02) [inline addon/services/intl.js]: :+1:

File: addon/services/intl.js (hunk 1 above)
Comment (elwayman02) [inline addon/services/intl.js]: What is the current signature of the `{{t}}` helper? I'd like to maintain the same arguments that it supports.

Comment (jasonmit) [inline addon/services/intl.js]: `{{t 'foo' bar='bar' locale='fr'}}`

The formats gets threaded through automatically.  But when invoking it programmatically, you have the ability to thread a format POJO.  Ultimately, this gets invoked: https://github.com/yahoo/ember-intl/blob/master/addon/formatters/format-message.js#L21 and that all should be satisfied

File: addon/services/intl.js [hunk 3]
```diff
@@ -52,6 +52,10 @@ const IntlService = Service.extend(Evented, {
   formatTime: formatterProxy('time'),
   formatDate: formatterProxy('date'),
 
+  t(key, options, formats) {
+    return this.formatMessage(this.findTranslationByKey(key, options.locale), options, formats);
```
Comment (jasonmit) [inline addon/services/intl.js]: will throw if options is undefined on options.locale.  it's optional

File: addon/services/intl.js [hunk 4]
```diff
@@ -52,6 +52,10 @@ const IntlService = Service.extend(Evented, {
   formatTime: formatterProxy('time'),
   formatDate: formatterProxy('date'),
 
+  t(key, options, formats) {
+    return this.formatMessage(this.findTranslationByKey(key, options ? options.locale : undefined), options, formats);
```
Comment (jasonmit) [inline addon/services/intl.js]:
```js
let translation = this.findTranslationByKey(key, options && options.locale);

return this.formatMessage(translation, options, formats);
```

I would rather split this up for debugging purposes

Comment (elwayman02) [inline addon/services/intl.js]: ok

File: tests/unit/services/intl.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
+import { moduleFor } from 'ember-qunit';
+import test from 'dummy/tests/ember-sinon-qunit/test';
+
+let service;
+
+moduleFor('service:intl', 'Unit | Service | intl', {
```
Comment (jasonmit) [inline tests/unit/services/intl.js]: I'd rather see more of an integration style test to get the whole flow.  I'm fine with keeping these unit tests though.

Comment (elwayman02) [inline tests/unit/services/intl.js]: I agree, but I don't have time to address that today. Is this sufficient to merge the changes and we can log an issue to provide better overall unit/integration tests for the service?

Comment (jasonmit) [inline tests/unit/services/intl.js]: Okay, will hold off on versioning though.

Comment (elwayman02) [inline tests/unit/services/intl.js]: :+1:

**Pull request merged**